### ISSUE TITLE
"tabulator is not a function" in node application

### ISSUE THREAD

**Issue by bmc0877** [olifolkerd/tabulator#310]
I'm using require to bring in jquery, jquery-ui-dist and jquery.tabulator.  I can create jquery-ui widgets like this:

$("#slider").slider();

But this gives me an error:

$("#tabulator").tabulator();

Any ideas?  Or code examples?  Thanks!

**Comment by olifolkerd** [comment]
That would suggest that either you are including your tabulator.js before the jQuery libraries, or that you are trying to create a tabulator before you have included the tabulator.js.

Also you need to pass a constructor object to your tabulator to create it, eg:
```js
$("#tabulator").tabulator({
    columns:[
        {title:"name", field:"name"),
   ]
});
```

**Comment by bmc0877** [comment]
I'm using require like this:
```
const $ = require('jquery');
global.jQuery = $;
const jqueryui = require('jquery-ui-dist/jquery-ui');
const jquerytabulator = require('jquery.tabulator');
```

**Comment by olifolkerd** [comment]
I don't think you need to assign jquery ui or tabulator to variables:

```js
var $ = require('jquery');
require('jquery-ui-dist/jquery-ui');
require('jquery.tabulator');
```

let me know if that helps

Cheers

Oli

**Comment by bmc0877** [comment]
Thanks for the reply, but this still doesn't work for me using requireJs.  So instead, I'm leaving my <script> tags in the html file for jquery, jquery-ui-dist and jquery.tabulator and in my javascript module I removed all 3 require lines and replaced them with this:
`const $ = global.jQuery;`
Maybe not the best solution, but it's working.

**Comment by olifolkerd** [comment]
Thanks for letting me know, i will do some more digging and add the recommended process to the documentation.

**Comment by olifolkerd** [comment]
I have just merged in a pull request that deals with this #280 

Download the latest version from the repo and hopefully you should be sorted.

Cheers

Oli

**Comment by olifolkerd** [comment]
Hey,

Great news, i finally got to the bottom of the node issues! there was an issue with the package.json ***main*** property pointing to the wrong file.

I have just released version 3.1.1 to fix the issue.

Cheers

Oli

**Comment by bmc0877** [comment]
Great, thanks!
